common/ferrostar/src/routing_adapters/algorithms.rs

```diff
@@ -0,0 +1,26 @@
+use polyline::decode_polyline;
+
+use crate::models::GeographicCoordinate;
+
+use super::error::ParsingError;
+
+/// Parse a polyline-encoded geometry string into a list of geographic coordinates.
+/// If the polyline cannot be decoded, a [`ParsingError`] results.
+pub fn get_coordinates_from_geometry(
```

Comment:
Did this move from somewhere like the main `algorithms` module? If so, I think I probably put it there and that it became a bit of a kitchen sink :P

While might belong under the `routing_adapters` module (honestly not sure), it's more of a common utility conversion function than any algorithm.

Doesn't need to happen for this PR since we're still pre-1.0 and can always change it, but there's probably a better name for the module.

Reply:
Yeah, definitely worth chatting about and maybe doing a follow up ticket on organization. This was exactly the consideration, the root level `algorithms.rs` was going to become pretty crazy.